doc/developer_guide/developer_guide.md

```diff
@@ -0,0 +1,121 @@
+
+```bash
+luarocks path >> ~/.bashrc
+```
+
+### Installing the Required Lua Packages
+
+You need the packages for unit testing, mocking and JSON processing.
+
+Execute as `root` or modify to install in your home directory:
+
+```bash
+sudo luarocks install LuaUnit
+sudo luarocks install Mockagne
+sudo luarocks install lua-cjson
+sudo luarocks install remotelog
+sudo luarocks install luacov
+sudo luarocks install luacov-coveralls
+sudo luarocks install luacheck
+```
+
+Most of those packages are only required for testing. While `cjson` is needed at runtime, it is prepackaged with Exasol, so no need to install it at runtime.
+
+The `luacov` and `luacov-coveralls` libraries take care of measuring and reporting code coverage in the tests.
+
+## How to Run Lua Unit Tests
+
+### Run Unit Tests From Terminal
+
+To run unit tests from terminal, you first need to install Lua:
+
+```bash
+sudo apt install lua5.1
+```
+
+Another important thing to do, you need to add the project's directories with lua files to LUA_PATH environment variable.
+We add two absolute paths, one to the `main` and another to the `test` folder: 
+
+```bash
+export LUA_PATH='/home/<absolute>/<path>/row-level-security-lua/src/main/lua/?.lua;/home/<absolute>/<path>/row-level-security-lua/src/test/lua/?.lua;'"$LUA_PATH"
+```
+
+After that you can try to run any test file:
+
+```bash
+lua src/test/lua/exasolvs/test_query_renderer.lua 
+```
+
+If you want to run all unit tests including code coverage and static code analysis, issue the following command:
+
+```bash
+tools/runtests.sh
+```
+
+The test output contains summaries and you will find reports in the `luaunit-reports` and `luacov-reports` directories.
+
+### Understanding the Sources
+
+Under [doc/model](../../model) you find a UML model of the project that you can render with [PlantUML](https://plantuml.com/). We recommend studying the model to understand structure and behavior.
+
+You can render the model by running:
+
+```bash
+mvn com.github.jeluard:plantuml-maven-plugin:generate
+```
+
+The resulting SVG files are located under [target/plantuml](../../target/plantuml). They contain links for drilling down.
+
+Since the model contains all imporant information, here just a very short summary.
+
+1. VSCL provides a base library for writing your own Virtual Schemas in Lua
+1. The resulting package is available as LuaRocks package `virtual-schema-common-lua`
+1. In your concrete Virtual Schema implementation you need to write an `entry` module, that has an `adapter_call` entry function
+1. The `entry` module should create and wire up all static objects (like the `RequestDispatcher` for example)
+1. Use the `remotelog` package for logging
+
+### Running the Unit Tests From Intellij IDEA
+
+First, you need to install a plug-in that handles Lua code. We recommend to use `lua` plugin by `sylvanaar`.
```

Comment:
I thought it was EmmyLua?

Reply:
It is. I will change that in another ticket: #5